components/net/fetch/methods.rs

```diff
@@ -317,7 +320,32 @@ fn basic_fetch(request: Rc<Request>, cache: &mut CORSCache) -> Response {
             }
         },
 
-        "blob" | "file" | "ftp" => {
+        "file" => {
+            if *request.method.borrow() == Method::Get {
+                match url.to_file_path() {
+                    Ok(file_path) => {
+                        let file_path_dup = &file_path.to_owned();
+                        if let Some(mut file) = File::open(file_path).ok() {
```

Comment:
Also, can this use `file_path.clone()` instead? That way you can get rid of `file_path_dup`.